app/src/main/java/com/spascoding/taskycourse/feature_agenda/presentation/detail_screen/DetailViewModel.kt

```diff
@@ -33,19 +42,19 @@ class DetailViewModel @Inject constructor() : ViewModel() {
         }
     }
 
-    fun setNavigation(navigation: Navigation) = when (navigation) {
-        Navigation.EventDetailNavigation -> {
-            _state.value = state.value.copy(title = R.string.new_event)
-            _state.value = state.value.copy(description = R.string.event_description)
-        }
-        Navigation.RemainderDetailNavigation -> {
-            _state.value = state.value.copy(title = R.string.new_reminder)
-            _state.value = state.value.copy(description = R.string.reminder_description)
-        }
-        Navigation.TaskDetailNavigation -> {
-            _state.value = state.value.copy(title = R.string.new_task)
-            _state.value = state.value.copy(description = R.string.task_description)
+    fun setNavigation(navigation: Navigation) {
+        savedStateHandle.set("navigation", navigation.route)
+        when (navigation) {
+            Navigation.EventDetailNavigation -> {
+                _state.value = state.value.copy(title = R.string.new_event, description = R.string.event_description)
```

Comment:
In general, I'd always use `_stateFlow.update { it.copy(...) }` to avoid race conditions. This is how it could happen:

*StateFlow holds value A*
- Thread 1 calls `copy()` on StateFlow with value A
- Before Thread 1 assigns its changed copy, Thread 2 updates StateFlow with value B
- Now, Thread 1 assigns its changed copy, which is still based on value A though, but the latest value is already value B. Therefore, value B would never be considered here.

Not always an issue, but I worked on highly stateful apps with frequently changing states, in which this really bit me, since such bugs are super hard to find.
Using `_state.value = _state.value.copy(...)` is fine if you always call this from a single thread (which you do here). But in ViewModels, you're very likely to not do that as soon as you'd change the state inside a `viewModelScope`.